Conv updates

dverbeir opened the pull request.

dverbeir committed:
add Magic Number as per RFC

Signed-off-by: David Verbeiren <david.verbeiren@tessares.net>

dverbeir committed:
add error codes from RFC8803

Signed-off-by: David Verbeiren <david.verbeiren@tessares.net>

dverbeir committed:
convert_util: expose CONVERT_ALIGN() macro

This macro can be useful for client code to allocate space for
Convert messages as the space should indeed account for the 4-bytes
alignment. This would be the case for example when calling
convert_write(): the function would return an error if the provided
buffer is not large enough for the 4-bytes alignment.

Signed-off-by: David Verbeiren <david.verbeiren@tessares.net>

GregoryVds commented:
Hi David,
Unfortunately I think we need to bump Ubuntu to 20.x in the Dockerfile as 19.04 is EOF.

matttbe commented on convert.h:
```diff
@@ -38,6 +38,7 @@
 #include <netinet/in.h>
 
 #define CONVERT_VERSION 1
+#define CONVERT_MAGIC_NO 0x2263
```
or 8803 in decimal format :)
(like the RFC)
But best to keep 0x2263 as specified on the RFC

matttbe reviewed (approved):
Looks good to me, good job!

GregoryVds reviewed (approved):
Thanks for adding this!

dverbeir committed:
Ubuntu 19.04 is EOL but keep using it

We cannot move right-away to Ubuntu 20.04 because
* lib syscall_intercept doesn't work with more recent libc
  See issue https://github.com/pmem/syscall_intercept/issues/97.
  Proposed change is not merged yet (though appears to work,
  so we could go to that specific commit)
* scapy operation seems to have issues where an ACK goes out
  automatically before our SYN+ACK response to the incoming SYN.

Also, there are other problems building scapy on 19.10.

Let's differ the move to 20.04 LTS to a later time.

Signed-off-by: David Verbeiren <david.verbeiren@tessares.net>

GregoryVds commented:
I think you need something like this in the Dockerfile:
`ENV DEBIAN_FRONTEND noninteractive`

Docker build is blocked on the following:
```Please select the geographic area in which you live. Subsequent configuration
questions will narrow this down by presenting a list of cities, representing
the time zones in which they are located.
  1. Africa      4. Australia  7. Atlantic  10. Pacific  13. Etc
  2. America     5. Arctic     8. Europe    11. SystemV
  3. Antarctica  6. Asia       9. Indian    12. US
Geographic area: ```

GregoryVds commented:
> I guess this will fail, e.g. clang-tools-6.0 is not available in 20.04. Can we install any version of CLang?

No idea :-) Let's see what is says with another version :crossed_fingers:

GregoryVds commented:
:'( New cppcheck is apparently catching new lint errors

dverbeir committed:
convert_client: cppcheck doesn't like LIST_HEAD(, ...

Just give the struct a name to make cppcheck happy.

Signed-off-by: David Verbeiren <david.verbeiren@tessares.net>

dverbeir commented:
@GregoryVds Can we merge this PR now? I forced usage of EOL'd Ubuntu 19.04 for now as you suggested.

GregoryVds commented:
@dverbeir Sure, looking good to me!
Sorry, I didn't notice that you pushed that last update on the Dockerfile
